parsexml.py

```diff
@@ -9,39 +9,82 @@
 #structure of the document, then transforms the document to json and saves it
 #to a file.
 
-import sys, json, xmltodict, os, logging, time, argparse, glob
+import sys, json, xmltodict, os, logging, time, argparse, glob, requests
+import dateutil.parser
+
 from datetime import datetime
 
+
 #change extension of file name to specified extension
 def change_ext(fname, ext):
     seq = (os.path.splitext(fname)[0], ext)
     return '.'.join(seq)
 
+#get field(date) and return in a proper iso format
+def format_date(field,x):
+    date = dateutil.parser.parse(x.pop(field))
+    dateiso = date.isoformat()+'Z'
+    return dateiso
+
 #this function contains the code for parsing the xml file
 #and writing the results out to a json file
 def processFile(fname):
+    type = os.path.splitext(fname)[1]
+    doc = {}
     try:
         with open(os.path.abspath(fname)) as fd:
-            fn = change_ext(fname,'json')
-            doc = xmltodict.parse(fd.read())
-
+            if type == ".xml":
+                fn = change_ext(fname,'json')
+                doc = xmltodict.parse(fd.read())
+            elif type == ".json":
+                doc = json.loads(fd.read())
             #get document id from metadata xml file
-            for x in doc['main']['DATA_RECORD']:
-                x['appid'] = x.pop('BD_PATENT_APPLICATION_NO')
-                x['doc_date'] = x.pop('DOCUMENT_CREATE_DT')
-                value = x.get('DOCUMENT_IMAGE_ID',x.get('DOCUMENT_NM'))
-                txtfn = os.path.join(os.path.dirname(fn),'PDF_image',value+'.txt')
-                if os.path.isfile(txtfn):
-                    with open(txtfn) as dr:
-                        text = dr.read()
-                        x['textdata'] = text
-                else:
-                    logging.error("PDF text file does not exist. JSON file creation skipped.")
-                    return
+            records = doc['main']['DATA_RECORD']
+            for x in records:
+                docid = x.get('DOCUMENT_IMAGE_ID',x.get('DOCUMENT_NM'))
+                if type == ".xml":
+                    txtfn = os.path.join(os.path.dirname(fn),'PDF_image',docid+'.txt')
+                    if os.path.isfile(txtfn):
+                        x['LAST_MODIFIED_TS'] = format_date('LAST_MODIFIED_TS',x)
```

Comment:
Just a stylistic thing... Usually we pass in the object/record as the first argument, then the more specific info after that, so I would reverse the argument order here, e.g.:

`format_date(x, 'LAST_MODIFIED_TS')`